scripts/dashboards_ui_lib.py

```
@@ -0,0 +1,389 @@
+    deadline = time.time() + timeout_seconds
+    while time.time() < deadline:
+        matches = find_index_patterns(base_url, title)
+        if matches:
+            return matches
+        time.sleep(1)
+    raise RuntimeError(
+        f"timed out waiting for Dashboards to save index pattern {title!r}"
+    )
+
+
+def save_failure_screenshot(driver: Any, path: str, logger: Logger) -> None:
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `driver`

(ANN401)